mantidproject/mantid
# Add option for number of cylinder slices to SolidAngle

Add option to specify how many angular slices will be used when triangulating a cylinder in order to calculate solid angle. Do this by passing a new object - ``SolidAngleParams`` - around when calculating solid angle. There is also a new argument defined on the actual ``SolidAngle`` algorithm. The default value will be 10, which is what is used currently, so no results should change. To fix issue #36493 we'll need to change the default value.

#### Summary of work

Added a new type, `SolidAngleParams` that holds the arguments required for performing the solid angle calculation. Most of the changes in the code are plumbing to pass this object around. The actual calculation happens in `CSGObject.cpp`.

See #36493 for the problem that this relates to. 




### To test:
No results should change when calling `SolidAngle`, unless you use the new argument called `NumberOfCylinderSlices`. To see results from this you can use the following:
```
ws = LoadEmptyInstrument(InstrumentName = "SANS2D")
sa = SolidAngle(ws, NumberOfCylinderSlices=13)
```
Then look at the instrument view (or the data) for the `sa` workspace.







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
